### PR TITLE
[WIP] Quick Reply Modal

### PR DESCRIPTION
**Pull request by aliseyalvi**
### What does this PR?
Added basic modal for adding quick comment to a post in feeds screen.
Needs UI refinement and method to broadcast the comment.

### Issue number
Trello: Tapping on comment icon for quick replies on feed

### Screenshots/Video
https://www.loom.com/share/31b52b803c5942ec9a79ce8ec2641604

### CODE REVIEW

**Comment by aliseyalvi** [comment]
Refined UI and added two step quick reply https://github.com/ecency/ecency-mobile/pull/2138/commits/1960c5dfa5980cce65bca28d90ea67fb2d9e788d.

https://www.loom.com/share/b1b46f763b8a4705941952c33823d53b

**Comment by aliseyalvi** [comment]
Done UI changes .
Changed to 1 step reply.
https://www.loom.com/share/e540b5082e3b46c7a67cb3b64b8a3c7e

**Comment by noumantahir** [comment]
@aliseyalvi  make sure testing modal in dark mode as well... post images of android, ios, light and dark mode... if possible do test on small sized device as well like iPhone mini or 6 and if android emulator is active try some < 5 inch device like nexus 5 may be.

**Comment by aliseyalvi** [comment]
@noumantahir  

- upgraded react-native-actions-sheet version which fixes the alignment issue of modal https://github.com/ecency/ecency-mobile/pull/2138/commits/1b229f4c9ead61ae543f50fe561f9aa8bd54c5ad.
- removed loadPosts method.
- checked in dark mode and on iphone6/nexus5 and small screen devices. Looks good to me.

Output on iphone 11 and pixel 3:
Android Emulator Pixel 2 API 295554


Output in dark mode:
Android Emulator Pixel 2 API 295554

Output on iphone 6 and nexus 5 (small screen devices);
Android Emulator Nexus 5 API 295554

**Review by noumantahir — CHANGES_REQUESTED** [review]
@aliseyalvi 
Check screen shot detailed feedback markedup.
Screenshot 2022-01-13 at 1 39 07 PM

**Comment by noumantahir** [comment]
@aliseyalvi modal is still not very intuitive for logged out user... perhaps better if we not show the modal at all may be... @feruzm your suggestion...?

Right now user is presented with this error on reply as there is no user.
Screenshot 2022-01-13 at 10 30 15 PM

File: src/components/quickReplyModal/quickReplyModalView.tsx [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+        currentAccount,
+        pinCode,
+        parentAuthor,
+        parentPermlink,
+        permlink,
+        commentValue,
+        parentTags,
+      )
+        .then(() => {
+          AsyncStorage.setItem('temp-reply', '');
```
Comment (noumantahir) [inline src/components/quickReplyModal/quickReplyModalView.tsx:125]: We do not need to clear `temp-reply` here as we are not making use of temp-reply in quick reply modal, it is only being used for full comment editor.

Comment (noumantahir) [reply src/components/quickReplyModal/quickReplyModalView.tsx:125]: @aliseyalvi rather what we should do is clear text state, the attached screenshot show same comment after it has been successfully made, it should appear empty here....

Screenshot 2022-01-13 at 10 47 50 PM

File: src/components/quickReplyModal/quickReplyModalView.tsx [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        <Text style={styles.name}>{`@${currentAccount.username}`}</Text>
+      </View>
+    </View>
+  );
+
+  const _renderContent = () => {
+    return (
+      <View style={styles.modalContainer}>
+        {_renderSummary()}
+        {isLoggedIn && _renderAvatar()}
```
Comment (noumantahir) [inline src/components/quickReplyModal/quickReplyModalView.tsx:177]: Not sure if this does any good for user experience, do check the screenshot shared in conversation thread of this PR for more details.

**Review by noumantahir — CHANGES_REQUESTED** [review]

**Comment by noumantahir** [comment]
@aliseyalvi swipe down to close gesture not working on android, do check if that can be easily resolved, otherwise may be adding a 'X' button somewhere on modal might help as well.

https://user-images.githubusercontent.com/6298342/149383420-fa1db1c2-0f3f-4415-8d2f-1f0ce4cf3c8a.mov

**Comment by feruzm** [comment]
@noumantahir, @aliseyalvi let's not show comment modal if no user is logged in, better forward them to login page.

**Comment by aliseyalvi** [comment]
> @aliseyalvi swipe down to close gesture not working on android, do check if that can be easily resolved, otherwise may be adding a 'X' button somewhere on modal might help as well.
> 
>  Screen.Recording.2022-01-13.at.10.54.36.PM.mov

@noumantahir gestureEnabled prop is set on action sheet but it deosn't work due to keyboard is opened. It does work when keyboard is closed. I'll add a close icon which will close the modal and keyboard both but on iOS side keyboardShouldPersistTaps props is not working and tapping on sheet closes the keyboard and then we have to close the sheet.

**Comment by aliseyalvi** [comment]
@noumantahir  https://github.com/ecency/ecency-mobile/pull/2138/commits/d385404d0bfea37b03c01c7bef6b574c3339d494

- fixed the keyboardShouldPersistTaps issue by adding react-native-portalize and wrapping the component with portal.
- added close icon to close the modal and keyboard at same time
- done the mentioned changes in PR conversation

Output:
image

File: src/components/quickReplyModal/quickReplyModalView.tsx (hunk 2 above)
Comment (aliseyalvi) [reply src/components/quickReplyModal/quickReplyModalView.tsx]: hide modal if not logged in

File: src/screens/application/index.js [hunk 3]
```diff
@@ -62,15 +63,17 @@ const Application = () => {
...
-              <ApplicationScreen
-                isConnected={isConnected}
-                locale={locale}
-                toastNotification={toastNotification}
-                isReady={isReady}
-                isDarkTheme={isDarkTheme}
-                rcOffer={rcOffer}
-                foregroundNotificationData={foregroundNotificationData}
-              />
+              <Host>
```
Comment (noumantahir) [inline src/screens/application/index.js:66]: @aliseyalvi may be we should place <Host /> in `src/index.js` That seems to be the appropriate place for app wide initialisation as intl and persisteGate is also initialised that.

**Review by noumantahir — CHANGES_REQUESTED** [review]

**Comment by noumantahir** [comment]
Screenshot 2022-01-17 at 9 43 10 PM

**Review by noumantahir — APPROVED** [review]

**Review by feruzm — APPROVED** [review]
lgtm

**Pull request merged**